src/display/api.js

```diff
@@ -2536,12 +2536,17 @@ class WorkerTransport {
   }
 
   saveDocument(annotationStorage) {
-    return this.messageHandler.sendWithPromise("SaveDocument", {
-      numPages: this._numPages,
-      annotationStorage:
-        (annotationStorage && annotationStorage.getAll()) || null,
-      filename: this._fullReader.filename,
-    });
+    return this.messageHandler
+      .sendWithPromise("SaveDocument", {
+        numPages: this._numPages,
+        annotationStorage:
+          (annotationStorage && annotationStorage.getAll()) || null,
+        filename: this._fullReader.filename,
+      })
+      .then(results => {
+        annotationStorage.setUnmodified();
+        return results;
+      });
```

Comment:
If `this.messageHandler.sendWithPromise("SaveDocument", { ... })` is ever rejected, the modified-state of the `AnnotationStorage` thus won't be reset.

Obviously you could argue that it shouldn't be reset in that case, since saving did *not* in fact succeed. However if saving failed once, it seems unlikely a new `saveDocument` call won't also fail. Hence I wonder if simply resetting the modified-state *unconditionally* here won't be generally preferable, in order to avoid repeatedly prompting the user if saving ever fails.

All-in-all, would it thus make sense to change this block to something like the following instead?
```suggestion
      .finally(() => {
        annotationStorage.resetModified();
      });
```